File: android/src/main/java/expo/modules/rnnativelistview/RNNativeListviewModule.kt
package expo.modules.rnnativelistview

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class RNNativeListviewModule : Module() {
  override fun definition() = ModuleDefinition {

    Name("RNNativeListview")

    View(RNNativeListviewView::class) {
    }
  }
}

File: ios/RNNativeListviewModule.swift
import ExpoModulesCore
import React

public class RNNativeListviewModule: Module {
	public func definition() -> ModuleDefinition {
		Name("RNNativeListview")
		
		View(RNNativeListviewView.self) {
			Prop("list_style", {(view: RNNativeListviewView, prop: String) in
				var style: UITableView.Style = .plain
				if (prop == "insetGrouped") {
					style = .insetGrouped
				} else if (prop == "grouped") {
					style = .grouped
				} else if (prop == "plain") {
					style = .plain
				}
				view.updateTableViewStyle(style: style)
			})
		}
	}
}
